.github/workflows/cd.yaml

```diff
@@ -38,14 +38,98 @@ jobs:
       - uses: actions/checkout@v1
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set current date as env variable $TODAY
+        run: echo "TODAY=$(date +'%y-%m-%d')" >> $GITHUB_ENV
       - env:
+          CLUSTER: zq-test
+          NAMESPACE: ci-${{ github.run_number }}
           ZQD_ECR_HOST: ${{ secrets.AWS_ECR_HOST }}
-          ZQD_DATA_URI: s3://zqd-demo-1/mark/zqd-meta
-          ZQD_K8S_USER: ci-master
-          ZQD_TEST_CLUSTER: zq-test.us-east-2.eksctl.io
+          ZQD_DATA_URI: s3://zq-ci-test/${{ env.TODAY }}.${{ github.run_number }}
         run: |
-          aws eks --region us-east-2 update-kubeconfig --name zq-test
-          kubectl config set-context --current --namespace=ci-master
+          set -x
+          aws eks --region us-east-2 update-kubeconfig --name $CLUSTER
+          kubectl create ns $NAMESPACE || :
+          kubectl config set-context --current --namespace=$NAMESPACE
+          make helm-uninstall
+          make helm-install-recruiter
+          make helm-install-root
+          make helm-install-worker
+  eks-test-zapi-1:
+    # Run smoke tests for zqd image prior to deploying to ci-master.
+    needs: eks-test-deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.15'
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set current date as env variable
+        run: echo "TODAY=$(date +'%y-%m-%d')" >> $GITHUB_ENV
+      - env:
+          CLUSTER: zq-test
+          NAMESPACE: ci-${{ github.run_number }}
+          SOURCE_LOG_1: s3://brim-sampledata/wrccdc/zeek-logs/files.log.gz
+          SPACE: files
+          ZQD_DATA_URI: s3://zq-ci-test/${{ env.TODAY }}.${{ github.run_number }}
```

Comment:
Note the new bucket used here: zq-ci-test is bucket with a short retention policy (3 days right now). A new directory is created for every build, so simultaneous builds will not collide while creating and querying archives. This creates S3 garbage fast but it will get cleared out by the retention policy.